Src/ManageMessageNamespace/AddNamespaceComponent.cs

```diff
@@ -1,164 +1,88 @@
 ﻿using System;
 using System.ComponentModel;
 using System.IO;
+using BizTalkComponents.Utils;
 using Microsoft.BizTalk.Component.Interop;
 using Microsoft.BizTalk.Message.Interop;
 using Microsoft.BizTalk.Streaming;
 
-// TODO: Should have better GUI name for exposed parameters
-
 namespace Shared.PipelineComponents.ManageMessageNamespace
 {
     [ComponentCategory(CategoryTypes.CATID_PipelineComponent)]
     [ComponentCategory(CategoryTypes.CATID_Any)]
     [System.Runtime.InteropServices.Guid("950C8198-9AAD-467E-BA9C-16AA080C7D7C")]
 
-    public class AddNamespaceComponent : IBaseComponent,
+    public partial class AddNamespaceComponent : IBaseComponent,
         Microsoft.BizTalk.Component.Interop.IComponent,
         IComponentUI,
         IPersistPropertyBag
     {
+        private const string XPathPropertyName = "XPath";
+        private const string NewNamespacePropertyName = "NewNamespace";
+        private const string NamespaceFormPropertyName = "NamespaceFormPropertyName";
+        private const string ShouldUpdateMessageTypeContextPropertyName = "ShouldUpdateMessageTypeContext";
+
+        [RequiredRuntime]
+        [DisplayName("New Namespace")]
+        [Description("The new namespace to set.")]
         public string NewNamespace { get; set; }
+        [RequiredRuntime]
+        [DisplayName("Should update messagetype context")]
+        [Description("Specifies wether the message type should be updated with the new namespace.")]
         public bool ShouldUpdateMessageTypeContext { get; set; }
+        [RequiredRuntime]
+        [DisplayName("Namespace form")]
+        [Description("0 = Unqualified, 1 = Qualified, 2 = Default")]
         public NamespaceFormEnum NamespaceForm { get; set; }
+        [DisplayName("XPath")]
+        [Description("The path to set namespace on.")]
         public string XPath { get; set; }
```

Comment:
This is an optional value that is only used if one wants to set namespace on nodes that are below the root node. Is it possible to add that information to text and to validation?